### PR TITLE
application lifecycle refactoring related changes

### PR DESCRIPTION
**Pull request by mikeshng**
clusteradm related changes for app lifecycle refactoring https://github.com/open-cluster-management-io/multicloud-operators-subscription/pull/35 

/hold

Signed-off-by: Mike Ng <ming@redhat.com>

### CODE REVIEW

File: pkg/cmd/install/addons/scenario/addons/appmgr/clusterrole.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: multicloud-operators
+  name: multicluster-operators
```
Comment (qiujian16) [inline pkg/cmd/install/addons/scenario/addons/appmgr/clusterrole.yaml:5]: change this to a more specific name. I think it is only used for subscription

Comment (mikeshng) [reply pkg/cmd/install/addons/scenario/addons/appmgr/clusterrole.yaml]: You are right. Done as suggested, changed all `multicluster-operators` references to `multicluster-operators-subscription`. Thanks.

**Comment by qiujian16** [comment]
/approve
/lgtm
/hold

hold in case you need other to review. And I would REALLY love to see placement replaces placementrule as a followup :)

**Comment by mikeshng** [comment]
> I would REALLY love to see placement replaces placementrule as a followup :)

Based on discussion with @xiangjingli and @jnpacker we should have a concrete target date for adopting placement API by this Friday.

> hold in case you need other to review.

There are no one else that needs to review but I did need to update a yaml file (crd_report.yaml). Please take a quick look again, thanks.

**Pull request merged**